### PR TITLE
Add $row_id column to faker tables

### PR DESCRIPTION
**Pull request by losipiuk**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerConfig.java [hunk 1]
```diff
@@ -45,7 +45,7 @@ public long getDefaultLimit()
         return defaultLimit;
     }
 
-    @Config("faker.default-limit")
+    @Config("faker.default-rowsCount")
```
Comment (nineinchnick) [inline plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerConfig.java:48]: I don't think we should be using camel case in config property names

File: plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerPageSource.java [hunk 2]
```diff
@@ -132,13 +139,32 @@ class FakerPageSource
...
                 .collect(toImmutableList());
         this.pageBuilder = new PageBuilder(types);
     }
 
+    private Generator getGenerator(
+            FakerColumnHandle column,
+            TupleDomain<ColumnHandle> constraint,
+            long offset)
+    {
+        if (column.kind() == FakerColumnHandle.Kind.SYNTHETIC && ROW_ID_COLUMN_NAME.equals(column.name())) {
```
Comment (nineinchnick) [inline plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerPageSource.java:152]: I'm not sure we need column kind right now if there's only one synthetics column.

**Review by nineinchnick — APPROVED** [review]

File: plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerConfig.java (hunk 1 above)
Comment (nineinchnick) [reply plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerConfig.java:48]: This change also requires updating the docs

Comment (losipiuk) [reply plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerConfig.java:48]: this one is a mistake - will revert

File: plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerConfig.java [hunk 3]
```diff
@@ -37,7 +37,7 @@ void testExplicitPropertyMappings()
     {
         Map<String, String> properties = ImmutableMap.<String, String>builder()
                 .put("faker.null-probability", "1.0")
-                .put("faker.default-limit", "10")
+                .put("", "10")
```
Comment (nineinchnick) [inline plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerConfig.java:40]: Revert this

File: plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerColumnHandle.java [hunk 4]
```diff
@@ -20,13 +20,19 @@
 import static java.util.Objects.requireNonNull;
 
 public record FakerColumnHandle(
+        Kind kind,
```
Comment (nineinchnick) [inline plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerColumnHandle.java:23]: It's unused, remove

File: plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerSplit.java [hunk 5]
```diff
@@ -15,5 +15,14 @@
...
+import static com.google.common.base.Preconditions.checkArgument;
+
 public record FakerSplit(long splitNumber, long rowsCount)
-        implements ConnectorSplit {}
+        implements ConnectorSplit
+{
+    public FakerSplit
+    {
+        checkArgument(splitNumber >= 0, "splitNumber is negative");
+        checkArgument(rowsCount >= 0, "rowsCount is negative");
```
Comment (wendigo) [inline plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerSplit.java:27]: > 0 ?

File: plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java [hunk 6]
```diff
@@ -259,6 +261,9 @@ public synchronized void setColumnComment(ConnectorSession session, ConnectorTab
         TableInfo oldInfo = tables.get(tableName);
         List<ColumnInfo> columns = oldInfo.columns().stream()
                 .map(columnInfo -> {
+                    if (ROW_ID_COLUMN_NAME.equals(columnInfo.handle().name())) {
+                        throw new IllegalArgumentException(String.format("Cannot set comment for %s column", ROW_ID_COLUMN_NAME));
```
Comment (wendigo) [inline plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java:265]: checkArgument ?

File: plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerSplit.java (hunk 5 above)
Comment (losipiuk) [reply plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerSplit.java:27]: Not sure - I think 0 may be needed for some cases.

File: plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java (hunk 6 above)
Comment (losipiuk) [reply plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java:265]: naah

**Pull request merged**